pyroomacoustics/transform/stft.py

```diff
@@ -681,7 +697,7 @@ def _overlap_and_add(self, x=None):
 " ---------------------------------------------------------------------------- "
 # Authors: Robin Scheibler, Ivan Dokmanic, Sidney Barthe
 
-def analysis(x, L, hop, win=None, zp_back=0, zp_front=0):
+def analysis(x, L, hop, win=None, zp_back=0, zp_front=0, precision='double'):
```

Comment:
Here, I think we should not provide the `precision` keyword. Instead, we infer it from `x`'s type. Namely, call `STFT` with `precision=x.dtype` on line 735.

Same comment apply to the `synthesis` function.